docs/conceptual/Bar-Memory.rst

```diff
@@ -0,0 +1,97 @@
+.. meta::
+   :description: Learn about BAR memory and ways to handle physical addressing limit in ROCm
+   :keywords: BAR memory, MMIO, GPU memory, Physical Addressing Limit, AMD, ROCm
+
+******************************
+BAR memory overview
+******************************
+BAR (Base Address Register) memory in PCIe defines and maps the memory-mapped input/output (MMIO) space required by a PCIe device for its resources such as registers or device memory. CPU and other devices in the system use BAR memory to access the resources of the PCIe devices. When the system boots, the SBIOS sets up the physical address space of the hardware platform. This also includes system memory and one or more MMIO apertures. 
+
+Handling physical address limits for BAR memory access
+===============================================================
+In general, there are two MMIO apertures: one set up below 4 GB physical address space for 32-bit compatibility, and one above 4 GB for devices that need more space. You can control the memory location of the high MMIO aperture with above 4 GB physical address space from the configuration options in the SBIOS. This enables you to configure the extended MMIO space to align with the physical addressing limit of the devices on the hardware platform if peer-to-peer (P2P) Direct Memory Access (DMA) is required. The physical addressing limit is important when data transfer is needed between the local memory of different devices through P2P DMA. It works only when one device can directly access the local BAR memory of another. If the BAR memory is above the physical addressing limit of the device, it will not be able to access the remote BAR. For example, if a PCIe device is limited to 44-bit of physical addressing, you should ensure that the MMIO aperture is set below 44-bit in the hardware platform address space.
+
+You need to be aware of the physical addressing limits of the devices when setting up additional MMIO apertures. There are two ways to handle this:
+
+* Ensure that the high MMIO aperture is within the physical addressing limits of the devices in the system. For example, if the devices have a 44-bit physical addressing limit, set the ``MMIOH Base`` and ``MMIO High size`` options in the BIOS such that the aperture is within the 44-bit address range. Also, ensure the ``Above 4G Decoding`` option is Enabled. 
+
+* Enable the Input-Output Memory Management Unit (IOMMU). When the IOMMU is enabled in non-passthrough mode, it will create a virtual IO address space for each device on the system. It also ensures that all virtual addresses created in that space are within the physical addressing limits of the device. The driver reports the physical addressing limits to the kernel. The kernel sets the IO virtual address space for the device according to the physical addressing limits.
+
+
+BAR configuration for AMD GFX9 and Vega10 GPUs
```

Comment:
This section is likely to go stale (and the Vega 10 is already pretty old), so maybe it should be removed from here and put somewhere with more visibility, like with installation.